helix-core/src/main/java/org/apache/helix/model/LiveInstance.java

```diff
@@ -130,6 +140,22 @@ public void setHelixVersion(String helixVersion) {
     _record.setSimpleField(LiveInstanceProperty.HELIX_VERSION.toString(), helixVersion);
   }
 
+  /**
+   * Gets the live instance's status.
+   */
+  public LiveInstanceStatus getStatus() {
+    return LiveInstanceStatus.valueOf(_record.getSimpleField(LiveInstanceProperty.STATUS.name()));
```

Comment:
As you said, "For backward compatibility, the status field in live instance will be removed once it goes back to NORMAL.", this method will throw NPE under normal status. I guess your current test does not cover this case.